### PR TITLE
[Spring Cloud] Support end-to-end TLS

### PR DESCRIPTION
**Pull request by zmssp**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 1]
```diff
@@ -1391,9 +1394,29 @@ def domain_bind(cmd, client, resource_group, service, app,
             thumbprint=certificate_response.properties.thumbprint,
             cert_name=certificate
         )
+    if enable_end_to_end_tls is not None:
```
Comment (ShichaoQiu) [inline src/spring-cloud/azext_spring_cloud/custom.py:1397]: how about `if enable_end_to_end_tls`

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 2]
```diff
@@ -1415,6 +1439,9 @@ def domain_update(cmd, client, resource_group, service, app,
...
```
Comment (ShichaoQiu) [inline src/spring-cloud/azext_spring_cloud/custom.py:1442]: how about `if enable_end_to_end_tls`

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 3]
```diff
@@ -1391,9 +1394,29 @@ def domain_bind(cmd, client, resource_group, service, app,
...
+    properties = models_20201101preview.AppResourceProperties(enable_end_to_end_tls=enable_end_to_end_tls)
+    app_resource = models_20201101preview.AppResource()
+    app_resource.properties = properties
+    app_resource.location = location
+
+    logger.warning("Set end to end tls for app '{}'".format(app))
+    poller = client.apps.update(
+        resource_group, service, app, app_resource)
+    while poller.done() is False:
+        sleep(APP_CREATE_OR_UPDATE_SLEEP_INTERVAL)
```
Comment (fengzhou-msft) [inline src/spring-cloud/azext_spring_cloud/custom.py:1417]: Why not use `poller.result()` to let the LRO finish?

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 4]
```diff
@@ -91,6 +91,7 @@ def load_arguments(self, _):
                    help='If true, assign endpoint URL for direct access.',
                    options_list=['--assign-endpoint', c.deprecate(target='--is-public', redirect='--assign-endpoint', hide=True)])
         c.argument('https_only', arg_type=get_three_state_flag(), help='If true, access app via https', default=False)
+        c.argument('enable_end_to_end_tls', arg_type=get_three_state_flag(), help='If true, enable end to end tls')
```
Comment (fengzhou-msft) [inline src/spring-cloud/azext_spring_cloud/_params.py:94]: Do you want to add preview tag for this property if swagger is preview?

Comment (zmssp) [reply src/spring-cloud/azext_spring_cloud/_params.py:94]: No, we need to GA this feature.

File: src/spring-cloud/azext_spring_cloud/custom.py (hunk 3 above)
Comment (zmssp) [reply src/spring-cloud/azext_spring_cloud/custom.py]: Fixed, using poller.result()

**Review by fengzhou-msft — APPROVED** [review]

**Review by ShichaoQiu — APPROVED** [review]

**Pull request merged**